### ISSUE TITLE
[OJS] migrate from 3.1.2 to 3.4.0.6 : lost decision in activity log

### ISSUE THREAD

**Issue by forgive38** [pkp/pkp-lib#10361]
**Describe the bug**
Before upgrade, you have the type of decision in the activity log of a submission 
for example:  	An editor decision (Accept Submission) for article 20 was recorded by ....

And now you have:
An editor decision (0) for article 20 was recorded by ....

**To Reproduce**
1. install OJS 3.4.0.6
2. import DB of ojs 3.1.2 from datasets
3. run upgrade
4. and check activity log of a submission

**What application are you using?**
 OJS

Maybe the solution is: 
```
diff --git a/schemas/eventLog.json b/schemas/eventLog.json
index 23bc3a1aba..f4c53ec5f5 100644
--- a/schemas/eventLog.json
+++ b/schemas/eventLog.json
@@ -42,7 +42,7 @@
                        ]
                },
                "decision": {
-                       "type": "integer",
+                       "type": "string",
                        "description": "The decision type identifier. One of the Decision::* constants.",
                        "validation": [
                                "nullable",
```
It works because before 'decision' was saved as string in event_log_settings
but I don't know the impact ...

Thank you

### Additional Information
There is additional problem related to localization of decisions that get stored in the `event_log_settings` for settings_name `decision` which points to the translated decision but only available in the journal primary language . so changing the locale does not fetch decision in intended locale as that is never get stored . This will require some migrations to update current decisions data and need to update the mechanism to store and fetch localized decision data . 

<img width="3060" height="1912" alt="Image" src="https://github.com/user-attachments/assets/094884a4-934c-435f-8858-a86cdc01258d" />

some implementation approach(which fixed the issue) details at https://github.com/pkp/pkp-lib/issues/10361#issuecomment-3414730892 and https://github.com/pkp/pkp-lib/issues/10361#issuecomment-3520399314

#### Stable-3.4.0
[pkp-lib](https://github.com/pkp/pkp-lib) --> https://github.com/pkp/pkp-lib/pull/11940
[ojs](https://github.com/pkp/ojs) --> https://github.com/pkp/ojs/pull/5143 [TEST ONLY]

#### Stable-3.5.0
[pkp-lib](https://github.com/pkp/pkp-lib) --> https://github.com/pkp/pkp-lib/pull/11934
[ojs](https://github.com/pkp/ojs) --> https://github.com/pkp/ojs/pull/5141 [TEST ONLY]

#### main
[pkp-lib](https://github.com/pkp/pkp-lib) --> https://github.com/pkp/pkp-lib/pull/11941
[ojs](https://github.com/pkp/ojs) --> https://github.com/pkp/ojs/pull/5144 [TEST ONLY]

**Comment by marcusburghardt** [comment]
I also noticed this issue. It seems a low-hanging-fruit but I am new to this project and not sure how to collaborate on this or who to ping. Maybe @asmecher @touhidurabir @defstat @jardakotesovec @taslangraham can help?

**Comment by asmecher** [comment]
Hi @marcusburghardt! I'm not convinced it's low-hanging fruit -- the upgrade scripts are a lot to jump into as a first contribution to the project -- but you're welcome to give it a look. I'd suggest working through upgrades release by release, looking in the `event_log` table (it may have been called something different back in 3.1.2) to see where the entry disappears.

**Comment by touhidurabir** [comment]
Re-opening this one as this is not a duplicate of [Decision lost on some stages at upgrade time from pre 3.4.0 #11241](https://github.com/pkp/pkp-lib/issues/11241) as the decisions does not getting lost but properly preserved in `edit_decisions` table but issue lies with data fetching and schema cast process .

**Comment by ajnyga** [comment]
Thanks @touhidurabir for this swift fix. The good thing is that no data is actually lost and already having the decision in some language is a lot better than seeing the (0). In most cases this will go unnoticed because the editors are most likely working with the journal primary language.

**Comment by touhidurabir** [comment]
After a discussion with @asmecher , we have decided that the only minimum fix will go in 3.4 and 3.5 as the PRs proposed at https://github.com/pkp/pkp-lib/issues/10361#issue-2503110344 . this will fix the issue of decision being properly displayed in the activity log in context's primary locale . 

we can think of a comprehensive one to have proper localized decision storing/fetching process for the main branch later in a different issue .

**Comment by touhidurabir** [comment]
@Vitaliy-1 @bozana can you please review the PRs at https://github.com/pkp/pkp-lib/issues/10361#issue-2503110344 .

**Comment by bozana** [comment]
Hi @touhidurabir, they look good to me, thanks a lot!

**Comment by touhidurabir** [comment]
Merged all the `pkp-lib` PRs for stable-3.4.0, stable-3.5.0 and main branch .

**Comment by touhidurabir** [comment]
@Tribunal33 ready for QA check .

**Comment by Tribunal33** [comment]
@touhidurabir I believe this passes QA.  Not sure what expectations are but they have improved.  Decisions can be seen in both primary and secondary languages.  The secondary language seems to be translations from primary language.

<img width="599" height="294" alt="Image" src="https://github.com/user-attachments/assets/0d5cb560-c2b5-42f7-bdd2-38d4d1c16311" />

  French version with English decisions.

<img width="716" height="316" alt="Image" src="https://github.com/user-attachments/assets/e193f385-7807-4299-acc5-b5e790384326" />

**Comment by touhidurabir** [comment]
> The secondary language seems to be translations from primary language.

@Tribunal33 yes , this is one issue that not possible to resolved with minimal changes and require a comprehensive DB migration . As the decisions are stored in the `event_log_settings` in translated form of primary lang, we have no simple way to have it translated in other lang . after a discussion over this, we decided to go with this minimal approach which works for most cases , see at https://github.com/pkp/pkp-lib/issues/10361#issuecomment-3414730892 .
